pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java

```diff
@@ -43,10 +46,16 @@ public class OpChainSchedulerService extends AbstractExecutionThreadService {
 
   private final OpChainScheduler _scheduler;
   private final ExecutorService _workerPool;
+  private final Cache<Long, Long> _cancelledRequests;
 
   public OpChainSchedulerService(OpChainScheduler scheduler, ExecutorService workerPool) {
+    this(scheduler, workerPool, QueryConfig.DEFAULT_SCHEDULER_RELEASE_TIMEOUT_MS);
```

Comment:
The scheduler release timeout is used for promoting queries from available state to running state. Can we use a different value here? Ideally that value should be a bit higher than the release timeout.

Also, you can use `expireAfterWrite` instead since that seems more intuitive in this case.